pip/client/client.go

```diff
@@ -0,0 +1,229 @@
+// Package client implements client for Policy Information Point (PIP) server.
+package client
+
+import (
+	"errors"
+	"math"
+	"net"
+	"sync/atomic"
+	"time"
+
+	"github.com/infobloxopen/themis/pdp"
+)
+
+var (
+	// ErrConnected occurs if method connect is called after connection has been
+	// established.
+	ErrConnected = errors.New("connection has been already established")
+
+	// ErrNotConnected occurs if method get is called before connection has been
+	// established.
+	ErrNotConnected = errors.New("connection hasn't been established yet")
+)
+
+// Client defines abstract PIP service client interface.
+type Client interface {
+	// Connect establishes connection to given PIP server.
+	Connect() error
+
+	// Close terminates previously established connection if any. Close silently
+	// returns if connection hasn't been established yet or if it has been
+	// already closed.
+	Close()
+
+	// Get requests information from PIP.
+	Get(args []pdp.AttributeAssignment) (pdp.AttributeValue, error)
```

Comment:
Would you consider returning `[]pdp.AttributeAssignment` instead of `pdp.AttributeValue`? It seems somewhat restricting as it stands.